src/System.Net.Http/src/Internal/Mail/MailBnfHelper.cs

```diff
@@ -330,9 +330,10 @@ internal static string GetDateTimeString(DateTime value, StringBuilder builder)
                 localBuilder.Append(' ');
             }
 
-            string[] offsetFields = offset.Split(':');
-            localBuilder.Append(offsetFields[0]);
-            localBuilder.Append(offsetFields[1]);
+            int colonIndex = offset.IndexOf(':');
+            int nextColonIndex = offset.IndexOf(':', colonIndex + 1);
+            localBuilder.Append(offset, 0, colonIndex);
+            localBuilder.Append(offset, colonIndex + 1, nextColonIndex - (colonIndex + 1));
             return (builder != null ? null : localBuilder.ToString());
```

Comment:
> Hadn't realized that

When making and submitting any such changes, please verify that the code is being appropriately tested.